core/src/main/scala/kafka/log/LogManager.scala

```diff
@@ -362,7 +401,7 @@ class LogManager(val logDirs: Array[File],
         val dataDir = nextLogDir()
         val dir = new File(dataDir, topicPartition.topic + "-" + topicPartition.partition)
         dir.mkdirs()
-        val log = new Log(dir, config, recoveryPoint = 0L, scheduler, time)
+        val log = new Log(dir, config, logStartOffset = 0L, recoveryPoint = 0L, scheduler, time)
```

Comment:
I am not sure I understand the problem of inconsistency here. I think this `createLog()` will only be called when the replica doesn't exist on the broker, which means the broker should not have logStartOffset for this replica, right? Are you considering the scenario that the replica is deleted (either through StopReplicaRequet or manually deleted via `rm -rf`) but the logStartOffset is still there? In that case I think we should actually ignore the outdated logStartOffset in the checkpoint file and read the value from leader. Because logStartOffset can be considered as logically deleted if log file is deleted. Does this make sense?